akka-projection-slick/src/main/scala/akka/projection/slick/SlickProjection.scala

```diff
@@ -73,27 +74,31 @@ object SlickProjection {
         override def process(envelope: Envelope): Future[Done] = {
 
           val offset = sourceProvider.extractOffset(envelope)
-          val handlerAction = delegate.process(envelope)
-
           sourceProvider match {
             case vsp: VerifiableSourceProvider[Offset, Envelope] =>
-              vsp.verifyOffset(offset) match {
-                case VerificationSuccess =>
-                  // run user function and offset storage on the same transaction
-                  // any side-effect in user function is at-least-once
-                  val txDBIO = offsetStore
-                    .saveOffset(projectionId, offset)
-                    .flatMap(_ => handlerAction)
-                    .transactionally
-                  databaseConfig.db.run(txDBIO).map(_ => Done)
-                case VerificationFailure(reason) =>
-                  logger.warning(
-                    "The offset failed source provider verification after the envelope was processed. " +
-                    "The transaction will not be executed. Skipping envelope with reason: {}",
-                    reason)
-                  Future.successful(Done)
-              }
-            case _ => Future.successful(Done)
+              // run user function and offset storage on the same transaction
+              // any side-effect in user function is at-least-once
+              val txDBIO = offsetStore
+                .saveOffset(projectionId, offset)
+                .flatMap(_ => delegate.process(envelope))
+                .flatMap { action =>
+                  vsp.verifyOffset(offset) match {
```

Comment:
I think we left some pass on the previous PR. Not all `SourceProviders` will be a `VerifiableSourceProvider`. It only make sense to impl it for the Kafka case. 

Although I may create my own and impl `VerifiableSourceProvider`, I don't have too. The result is that when using it here, we will just skip the whole DB operation.

Reply:
Oops, you're right.